### PR TITLE
Make tests compatible with Python 2

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: test/test_train_mnist.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def train_mnist():
     devices = [':{}'.format(n) for n in range(0, FLAGS.num_cores)]
     inputs = torch.zeros(FLAGS.batch_size, 1, 28, 28)
     target = torch.zeros(FLAGS.batch_size, dtype=torch.int64)
-    xla_model = xm.XlaModel(model, [inputs], loss_fn=F.nll_loss, target=target,
+    xla_model = xm.XlaModel(model, (inputs), loss_fn=F.nll_loss, target=target,
```
Comment (dlibenzi) [inline test/test_train_mnist.py]: This does not look like correct.
(input) == input
Does not create a tuple.
We should be OK with a list ([]) here.
Or tuple([inputs])

File: torch_xla_py/xla_model.py [hunk 2]
```diff
@@ -237,7 +238,7 @@ def xla_loss(loss_fn, output_xla_tensors, labels):
             flat_tensors[flat_index].requires_grad = True
             replica_outputs.append(flat_tensors[flat_index])
             flat_index += 1
-        losses.append(loss_fn(*replica_outputs, labels[i]))
+        losses.append(loss_fn(tuple(replica_outputs), labels[i]))
```
Comment (dlibenzi) [inline torch_xla_py/xla_model.py]: Hmm, ... the loss function does not take a tuple.

Comment (asuhan) [reply torch_xla_py/xla_model.py]: Ok, how to unpack it then?

Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: Leave as is?

Comment (asuhan) [reply torch_xla_py/xla_model.py]: Also, I think this function is actually dead, let me remove it.

Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: No please, leave it. We might need it for API extensions ...

Comment (asuhan) [reply torch_xla_py/xla_model.py]: I cannot leave it "as is", this is a syntax error in Python 2. So, what to do about it, especially without test coverage?

File: test/test_train_mnist.py (hunk 1 above)
Comment (soumith) [reply test/test_train_mnist.py]: the right way is `(inputs, )`

File: torch_xla_py/xla_model.py (hunk 2 above)
Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: That should work!

Comment (asuhan) [reply torch_xla_py/xla_model.py]: Nope:

```
  File "build/bdist.linux-x86_64/egg/torch_xla_py/xla_model.py", line 241
    losses.append(loss_fn((*(list(replica_outputs).append(labels[i])))))
                           ^
SyntaxError: invalid syntax
```

Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: That worked for me ...

```
Python 2.7.13 (default, Nov 24 2017, 17:33:09) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> def f(a, b):
...   print(a,b)
... 
>>> f(*(list([1,2])))
(1, 2)
>>> 
```

Comment (asuhan) [reply torch_xla_py/xla_model.py]: The same snippet works for me as well, but for some reason the Python parser chokes for the change we actually need.

File: test/test_train_mnist.py (hunk 1 above)
Comment (asuhan) [reply test/test_train_mnist.py]: We shouldn't need a tuple at all here, just `inputs` would suffice.

File: torch_xla_py/xla_model.py (hunk 2 above)
Comment (asuhan) [reply torch_xla_py/xla_model.py]: Also, doesn't list append return `None`?

Comment (soumith) [reply torch_xla_py/xla_model.py]: my bad, I didn't check the return type of list append

File: test/test_train_mnist.py (hunk 1 above)
Comment (asuhan) [reply test/test_train_mnist.py]: Actually no, `TestReplicatedSum` is the only one which would break. I'll try another way.

**Comment by asuhan** [comment]
PTAL, I think I've addressed the issues.

File: torch_xla_py/xla_model.py [hunk 3]
```diff
@@ -237,7 +244,8 @@ def xla_loss(loss_fn, output_xla_tensors, labels):
             flat_tensors[flat_index].requires_grad = True
             replica_outputs.append(flat_tensors[flat_index])
             flat_index += 1
-        losses.append(loss_fn(*replica_outputs, labels[i]))
+        replica_outputs_and_label = _append_label_to_tensor_list(replica_outputs, labels[i])
```
Comment (dlibenzi) [inline torch_xla_py/xla_model.py]: Line length check

Comment (asuhan) [reply torch_xla_py/xla_model.py]: I'm running `autopep8` and it doesn't do anything to lines longer than 80 chars, I'm not sure what's going on.

Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: Maybe it's a g3 rule.
Would be nice to keep line lengths under control IMHO, even w/out the support of external tools.

Comment (asuhan) [reply torch_xla_py/xla_model.py]: Ok, turns out it requires `--max-line-length 79` to be specified in order to enforce *any* line length, help is misguiding (says the default is 79). I'm going to send another PR which formats this file and tests as well.

**Pull request merged**